### ISSUE TITLE
UseOptionForeachNotPatMatch false positive?

### ISSUE THREAD

**Issue by benblack86** [HairyFotr/linter#57]
Can't write this code using foreach since it is using tailrec. Can't think of another way to write this without using null.

```
  @tailrec
  final def tick(name: String, time: FiniteDuration): Unit =
    Option(times.putIfAbsent(name, time)) match { // linter:ignore UseOptionForeachNotPatMatch
      case None =>
      case Some(oldValue) =>
        if (!times.replace(name, oldValue, oldValue + time)) {
          tick(name, time)
        }
    }
```
